### ISSUE TITLE
Allow expanded and contracted object literal form, but disallow hybrid

### ISSUE THREAD

**Issue by Francisc** [jscs-dev/node-jscs#1931]
Hello,

Is there a way to validate this:
```
var obj={
  a:1,
  b:2
};

//as well as:
obj={a:1,b:2};

//but disallow this:
obj={a:1,
  b:2};
```

I think `requirePaddingNewLinesInObjects` with an new option of `{"allExcept": ["sameLine"]}` might work, but I'm not sure that's the best way. And looking at other rules I can't see an alternative.

**Comment by Francisc** [comment]
Or maybe a rule requiring a newline char after opening curly brace and before closing curly brace?

**Comment by markelog** [comment]
> Or maybe a rule requiring a newline char after opening curly brace and before closing curly brace?

Sounds interesting! I would recommend though, using https://github.com/mafintosh/is-my-json-valid package if your options would became complicated

**Comment by Francisc** [comment]
Hm, not sure I understand what you mean by that link.
It seems to be a JSON validator and all 3 forms of `obj` I posted above are valid JSON.

**Comment by markelog** [comment]
I meant instead of using bunch of asserts in `configure` method, you can use json schema. It is not a stopper though, but something to help you out, you can do those checks with the old fashion way though

**Comment by Francisc** [comment]
I want to disallow a code-style choice rather than a code validity one.

**Comment by markelog** [comment]
Hm, nuh, that is not what i mean, but doing it old fashion way is also good

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
